lib/brute/http_basic_auth/engine.py

```diff
@@ -136,14 +143,22 @@ def start(target, users, passwds, ports, timeout_sec, thread_number, num, total,
         trying = 0
         keyboard_interrupt_flag = False
         for port in ports:
-            if check_auth(target, timeout_sec, language, port):
-                continue
+            if not target.startswith("http://"):
+                try:
+                    check_auth("https://"+target, timeout_sec, language, port, HEADERS)
+                    target = "https://" + target
+                except:
+                    pass
+            else:
+                if check_auth("http://"+target_to_host(target), timeout_sec, language, port, HEADERS):
+                    target = target
+
             for user in users:
                 for passwd in passwds:
                     t = threading.Thread(target=login,
                                          args=(
                                              user, passwd, target, port, timeout_sec, log_in_file, language,
-                                             retries, time_sleep, thread_tmp_filename, socks_proxy, scan_id, scan_cmd))
+                                             retries, time_sleep, thread_tmp_filename, socks_proxy, scan_id, scan_cmd, HEADERS))
```

Comment:
please do not pass headers as parameters as this is breaking the core framework - headers are not parameters which are passed from the core